src/app/pages/catalog/catalog.component.ts

```diff
@@ -5,22 +5,39 @@ import { CategoriesService } from '../../core/services/categories.service';
 import { AsyncPipe } from '@angular/common';
 import { CategoriesStoreService } from '../../core/services/categories-store.service';
 import { Observable } from 'rxjs';
+import { UiSpinnerComponent } from '../../shared/ui-kit/ui-spinner/ui-spinner.component';
+import { isLoading$ } from '../../store/categories.store';
+import { ErrorHandlerService } from '../../core/services/error-handler.service';
+import { UiErrorNotificationComponent } from '../../shared/ui-kit/ui-error-notification/ui-error-notification.component';
 
 @Component({
   selector: 'quiz-catalog',
   standalone: true,
-  imports: [ UiButtonComponent, UiQuizCardComponent, AsyncPipe ],
+  imports: [
+    UiButtonComponent,
+    UiQuizCardComponent,
+    AsyncPipe,
+    UiSpinnerComponent,
+    UiErrorNotificationComponent,
+  ],
   templateUrl: './catalog.component.html',
 })
 export class CatalogComponent implements OnInit {
   private readonly categoriesService = inject(CategoriesService);
   private readonly categoriesStoreService = inject(CategoriesStoreService);
+  private readonly errorHandlerService = inject(ErrorHandlerService);
 
   categories$!: Observable<any>;
```

Comment:
I believe you know the type of categories, let' update it here: categories$!: Observable<any>;